tools/loader/Loader.cpp

```diff
@@ -343,10 +345,12 @@ void Loader::compile(PlaceholderBindings &bindings) {
   cctx.mode = CompilationMode::Infer;
   if (emittingBundle()) {
     // Emit IR for the graph, compile it and save as a bundle.
-    EE_.save(F_, cctx, emitBundle, networkName);
+    ::glow::optimizeFunction(F_, *backend_, cctx);
+    backend_->save(F_, emitBundle, networkName);
   } else {
     // Emit IR for the graph and compile it.
-    EE_.compile(F_, cctx);
+    auto error = hostManager_->addNetwork(std::move(M_));
```

Comment:
Now after we `std::move` the module, `Function` object becomes unavailable. But it is still possible to call `loader.getFunction()`! This may or may not bring unexpected corruption.

For example, `ModelRunner.cpp` misbehaves when it's trying to `getFunction()->getName()`. Because this happens after compilation.

Reply:
Compilation is potentially destructive over the Function, and Partitioning definitely makes access to the original function make less sense. I think while it's convenient to keep using the Function after compilation it's a bad pattern we should remove.

Reply:
I agree that it's a bad pattern. But it's not an obvious pattern. How can we make it clear to everybody?

Reply:
@artemrakhov unique_ptr!

Reply:
I've removed unnecessary calls to getFunction that were just getting the name. I've also added comments to getFunction and getModule that they should not be called after compile.